src/Configuration/src/SpringBootBase/SpringBootEnvProvider.cs

```diff
@@ -14,59 +15,42 @@ namespace Steeltoe.Extensions.Configuration.SpringBoot
     /// <summary>
     /// Configuration provider that expands the contents of SPRING_APPLICATION_JSON's Spring-style '.' delimited configuration key/value pairs to .NET compatible form
     /// </summary>
-    public class SpringBootEnvProvider : ConfigurationProvider
+    public class SpringBootEnvProvider : JsonStreamConfigurationProvider
     {
-        internal ILogger<SpringBootEnvProvider> _logger;
-
         private const string SPRING_APPLICATION_JSON = "SPRING_APPLICATION_JSON";
-
-        internal string SpringApplicationJson { get; set; }
+        private readonly string _springApplicationJson;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="SpringBootEnvProvider"/> class.
         /// </summary>
-        /// <param name="logFactory">the logger factory to use</param>
-        public SpringBootEnvProvider(ILoggerFactory logFactory = null)
+        /// <param name="springApplicationJson"> The Json string to parse </param>
+        public SpringBootEnvProvider(string springApplicationJson = null)
+            : base(new JsonStreamConfigurationSource())
         {
-            _logger = logFactory?.CreateLogger<SpringBootEnvProvider>();
-            SpringApplicationJson = Environment.GetEnvironmentVariable(SPRING_APPLICATION_JSON);
+            _springApplicationJson = springApplicationJson;
         }
 
         /// <summary>
-        /// Expands and loads the new keys
+        /// Translates and loads the new keys
```

Comment:
```suggestion
        /// Maps SPRING_APPLICATION_JSON into key:value pairs
```
Just a thought... maybe `/// <inheritdoc />` makes the most sense though